packages/client/src/runtime/core/request/createPrismaPromise.ts

```diff
@@ -0,0 +1,54 @@
+import type { PrismaPromise } from './PrismaPromise'
+import type { Context } from '@opentelemetry/api'
+import { context } from '@opentelemetry/api'
+
+/**
+ * Creates a [[PrismaPromise]]. It is Prisma's implementation of `Promise` which
+ * is essentially a proxy for `Promise`. All the transaction-compatible client
+ * methods return one, this allows for pre-preparing queries without executing
+ * them until `.then` is called. It's the foundation of Prisma's query batching.
+ * @param callback that will be wrapped within our promise implementation
+ * @see [[PrismaPromise]]
+ * @returns
+ */
+export function createPrismaPromise(
+  callback: (transactionId?: number, runInTransaction?: boolean, otelCtx?: Context) => PrismaPromise<unknown>,
+): PrismaPromise<unknown> {
+  const otelCtx = context.active() // get the context at time of creation
```

Comment:
Capture the otel context here for the first time on creation of the prisma promise.